### PR TITLE
Check if current ArticleVersion is VOR and is a reviewed preprint

### PR DESCRIPTION
**Pull request by smoqadam**

### CODE REVIEW

**Review by nlisgo — CHANGES_REQUESTED** [review]
We also need to refactor the following code which currently uses the yaml file: https://github.com/elifesciences/search/blob/develop/src/Search/Console.php#L350-L365

File: bin/ci-reindex-reviewed-preprints [hunk 1]
```diff
@@ -4,6 +4,6 @@ set -e
 
 . bin/ci-start-processes
 
-./bin/console reviewedPreprint:reindex
+./bin/console queue:import ReviewedPreprints
```
Comment (nlisgo) [inline bin/ci-reindex-reviewed-preprints:7]: As we are no longer using the reviewed preprints config we can get rid of the `reviewedPreprint:reindex` command as it duplicates the above command.

File: src/Search/Console.php [hunk 2]
```diff
@@ -343,26 +341,15 @@ public function rdsReindexCommand(InputInterface $input, OutputInterface $output
...
-        $this->logger->info('Reviewed preprints added to indexing queue.');
-    }
-
     public function reviewedPreprintPurgeCommand(InputInterface $input, OutputInterface $output)
     {
         $this->logger->info('Purge reviewed preprints...');
         $ids = [];
-        foreach (array_keys($this->config['reviewed_preprints']) as $id) {
+        /** @var ApiSdk $sdk */
+        $sdk = $this->kernel->get('api.sdk');
```
Comment (nlisgo) [inline src/Search/Console.php:349]: Let's remove the need to call the api here and just purge all reviewed preprints from the elasticsearch. There may be a single command to do this or we can read them from the index and then delete each one in turn.

**Review by nlisgo — CHANGES_REQUESTED** [review]
Nearly there just one other optimisation on the purge command. Thanks!

File: src/Search/Console.php [hunk 3]
```diff
@@ -343,26 +342,15 @@ public function rdsReindexCommand(InputInterface $input, OutputInterface $output
...
-    }
-
     public function reviewedPreprintPurgeCommand(InputInterface $input, OutputInterface $output)
     {
         $this->logger->info('Purge reviewed preprints...');
         $ids = [];
-        foreach (array_keys($this->config['reviewed_preprints']) as $id) {
+        $query = new ElasticQueryBuilder($this->kernel->indexMetadata()->read());
+        $query = $query->whereType(['reviewed-preprint']);
+        $reviewedPreprints = $this->kernel->get('elastic.client.read')->searchDocuments($query->getRawQuery());
```
Comment (nlisgo) [inline src/Search/Console.php:351]: We should introduce a try catch here.

See: https://github.com/elifesciences/search/blob/develop/src/Search/Api/SearchController.php#L154-L168

File: src/Search/Console.php [hunk 4]
```diff
@@ -343,26 +342,15 @@ public function rdsReindexCommand(InputInterface $input, OutputInterface $output
...
-            $this->enqueue('article', $id);
-            $this->logger->info("Queuing reviewed preprint $id");
-            $this->enqueue('reviewed-preprint', $id);
-            $ids[] = $id;
-        }
-        $output->writeln('Queued: '.implode(', ', $ids).' (reviewed-preprint and article)');
-        $this->logger->info('Reviewed preprints added to indexing queue.');
-    }
-
     public function reviewedPreprintPurgeCommand(InputInterface $input, OutputInterface $output)
```
Comment (nlisgo) [inline src/Search/Console.php:345]: We could create a follow up ticket to create a single command to allow purging by type.

Comment (smoqadam) [reply src/Search/Console.php:345]: Ticket has been created: https://github.com/elifesciences/issues/issues/7985

**Comment by nlisgo** [comment]
There is a failure on CI with phpcs.

File: tests/src/Search/Workflow/ResearchArticleWorkflowTest.php [hunk 5]
```diff
@@ -85,10 +86,10 @@ public function testRdsDateIsUsedAsTheSortDateWhenThereIsAnRdsArticle()
     public function testReviewedDateAndCurationLabelsWhenThereIsAReviewedPreprint()
```
Comment (nlisgo) [inline tests/src/Search/Workflow/ResearchArticleWorkflowTest.php:86]: We can remove this test as we are not decorating the snippet.

**Review by nlisgo — APPROVED** [review]
Pending removal of test `testReviewedDateAndCurationLabelsWhenThereIsAReviewedPreprint`

**Review by nlisgo — APPROVED** [review]

**Pull request merged**